### ISSUE TITLE
get UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'toString' of null

### ISSUE THREAD

**Issue by mboveiri** [kubesail/deploy-to-kube#41]
Hi.
trying to `npx deploy-node-app ` and get `: " UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'toString' of null"`
the complete log : 

> npx: installed 120 in 22.963s
> (node:1196) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'toString' of null
>     at execSyncWithEnv (C:\Users\Administrator\AppData\Roaming\npm-cache\_npx\1196\node_modules\deploy-node-app\src\util.js:151:55)
>     at ensureBinaries (C:\Users\Administrator\AppData\Roaming\npm-cache\_npx\1196\node_modules\deploy-node-app\src\util.js:164:24)
>     at DeployNodeApp (C:\Users\Administrator\AppData\Roaming\npm-cache\_npx\1196\node_modules\deploy-node-app\src\deployNodeApp.js:509:30)
>     at Object.<anonymous> (C:\Users\Administrator\AppData\Roaming\npm-cache\_npx\1196\node_modules\deploy-node-app\src\index.js:38:1)
>     at Module._compile (internal/modules/cjs/loader.js:1147:30)
>     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1167:10)
>     at Module.load (internal/modules/cjs/loader.js:996:32)
>     at Function.Module._load (internal/modules/cjs/loader.js:896:14)
>     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
>     at C:\Program Files\nodejs\node_modules\npm\node_modules\libnpx\index.js:268:14
> (node:1196) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
> (node:1196) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by erulabs** [comment]
Hi @mboveiri - is it possible you could try `npx deploy-node-app@next` ?

**Comment by mboveiri** [comment]
> Hi @mboveiri - is it possible you could try `npx deploy-node-app@next` ?

try and get : `Command "which skaffold" failed to run: "spawnSync /bin/sh ENOENT"`

**Comment by erulabs** [comment]
Ack! Could you share what operating system you're using? Thanks!

**Comment by mboveiri** [comment]
I'm on Windows 10.

**Comment by erulabs** [comment]
Hey @mboveiri - Can you try ` npx deploy-node-app@next` again? Specifically, the fix is published in `2.0.12`. Let me know if that works for you!

**Comment by mboveiri** [comment]
thank you it worked but on the last step the error occurred : 

```
> What domain will this use? (leave blank for auto-generate)

Deploying to production!
(node:7812) UnhandledPromiseRejectionWarning: Error: spawnSync /bin/sh ENOENT
    at Object.spawnSync (internal/child_process.js:1045:20)
    at spawnSync (child_process.js:597:24)
    at execSync (child_process.js:642:15)
    at execSyncWithEnv (C:\Users\Administrator\AppData\Roaming\npm-cache\_npx\7812\node_modules\deploy-node-app\src\util.js:180:14)
    at DeployNodeApp (C:\Users\Administrator\AppData\Roaming\npm-cache\_npx\7812\node_modules\deploy-node-app\src\deployNodeApp.js:777:5)
(node:7812) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:7812) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by erulabs** [comment]
Ack! That error seems to imply there is no shell (/bin/sh) available on your system. If you're using Windows 10, is it possible you have the Linux Subsystem enabled?

My guess is there is no shell (sh or bash) on your system. I'll try to create a test for this and see if I can't figure out how to avoid needing bash or sh, but the easiest "fix" for you is to install something like [WSL](https://www.howtogeek.com/265900/everything-you-can-do-with-windows-10s-new-bash-shell/) or [git-scm](https://git-scm.com/download/win) both of which should provide you with a shell.

Let me know if you can provide any more information - I'd love to get this working for you - my only Windows machines have bash installed though, so it might take me a bit of time to sort out how to resolve this!

**Comment by mboveiri** [comment]
> Ack! That error seems to imply there is no shell (/bin/sh) available on your system. If you're using Windows 10, is it possible you have the Linux Subsystem enabled?
> 
> My guess is there is no shell (sh or bash) on your system. I'll try to create a test for this and see if I can't figure out how to avoid needing bash or sh, but the easiest "fix" for you is to install something like [WSL](https://www.howtogeek.com/265900/everything-you-can-do-with-windows-10s-new-bash-shell/) or [git-scm](https://git-scm.com/download/win) both of which should provide you with a shell.
> 
> Let me know if you can provide any more information - I'd love to get this working for you - my only Windows machines have bash installed though, so it might take me a bit of time to sort out how to resolve this!

i already have git-scm and run all command trough `git-bash` , and `sh` command run properly.
but for some reason can't test with WSL now, however thank you , i watch the updates for workaround.

**Comment by erulabs** [comment]
Oh! Interesting... Is it possible, in whatever shell you're running `deploy-node-app` in, could you run `echo $SHELL && which sh && which bash`  and paste the output here?

Thanks for your help and patience!

**Comment by mboveiri** [comment]
> echo $SHELL && which sh && which bash

Here is the response : 

```
/usr/bin/bash
/usr/bin/sh
/usr/bin/bash
```

**Comment by erulabs** [comment]
Hey @mboveiri - Thanks!! I've published `deploy-node-app@2.0.13` (also `deploy-node-app@next`) which should now properly figure out the path to `sh` on many different systems!

Thanks again for your help - hoping this version works for you - DNA2.0 is a work-in-progress, and it powers our repo builder at https://kubesail.com - so its slated to get lots and lots of love in the next few months! Please let us know if you run into any other issues!

**Comment by mboveiri** [comment]
Thank you it's work properly now, hope I was able to help the progress.
interest to new version and features.
